### ISSUE TITLE
NVDA reads 'Shortcut, Page 1' when reading every line in Word 2013

### ISSUE THREAD

**Issue by DanielVisionIreland** [nvaccess/nvda#16197]
### Steps to reproduce:
When arrowing down or using NVDA+A to read any Word 2013 document, NVDA announces "Shortcut, page 1" on every line navigated before announcing the text of the line.
### Actual behavior:

When arrowing down or using NVDA+A to read any Word 2013 document, NVDA announces "Shortcut, page 1" on every line navigated before announcing the text of the line.
### Expected behavior:

NVDA should not announce "Shortcut, page X" where X is the page number prior to each line.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2023.3
#### Windows version:
Windows 11 22H2
#### Name and version of other software in use when reproducing the issue:
Word 2013
#### Other information about your system:
Standard laptop
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes - Beta 2024.1B7
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by XLTechie** [comment]
A debug level log could be helpful here.

**Comment by Adriani90** [comment]
Are you using UIA in MS Word enabled?
Please go to advanced settings in NVDA settings dialog and set the UIA in MS Word value to "only when necessary". Is this issue then still reproducible?

**Comment by DanielVisionIreland** [comment]
> Are you using UIA in MS Word enabled? Please go to advanced settings in NVDA settings dialog and set the UIA in MS Word value to "only when necessary". Is this issue then still reproducible?

**Brilliant!** This is the solution! Many thanks Adriani, much appreciated. Greetings from Ireland :-)

**Comment by Adriani90** [comment]
You are welcome.
I am closing this issue for now since UIA is not supported in Office 2013 or 2016, but only in Office 365 and there this problem does not occur.

**Comment by XLTechie** [comment]
Why is UIA even being tried, for Word 2013? I thought our UIA support didn't 
engage for Word versions that old?

**Comment by lukaszgo1** [comment]
This issue should not be closed, since our code actively tries not to use UIA for versions of Office which are too old. If this is not working we should investigate why rather than ask affected users to reconfigure NVDA. @DanielVisionIreland Would you be able to set your logging level to debug in the general settings, reproduce the problem and upload the log file to this issue?

**Comment by Adriani90** [comment]
@lukaszgo1 support for Office 2013 ended completely in April 2023:
https://support.microsoft.com/en-gb/office/end-of-support-for-office-2013-90e4b0d1-098f-4656-b6e7-8b13b67ed62f

In Office 2016 I cannot reproduce this problem. Do you still think it is worthwhile investigating?
@DanielVisionIreland what is the reason you still stay with Office 2013 instead of updating to 2016 at least?

**Comment by lukaszgo1** [comment]
I do not particularly care about Office 2013 as such, however the code which forces UIA in recent enough builds of Office 2016 and later is very straight forward. If it does not work in some cases we should know why and improve it, to, for example, avoid enforcing UIA for non 365 versions of Office which are still in support. Looking at what can go wrong there it seems NVDA either fails to inject in process (which should work regardless of the Office version) or fails to retrieve version info from the executable (again that needs to be fixed regardless for which application it occurs).

**Comment by Adriani90** [comment]
@lukaszgo1 at this stage we don't really know if @DanielVisionIreland had this issue on a clean install of a NVDA or on a version that had UIA enabled explicitely.
@DanielVisionIreland which value had this UIA setting before you set it to "only when necessary"?

I think this is an edge case still and we need a new clear issue with clear str which shows that the default behaviour "where suitable" forces UIA on versions of MS Office older than 2016. Otherwise I don't see a way to investigate this further.

cc: @michaelDCurran for your information.

**Comment by XLTechie** [comment]
@Adriani90 it shouldn't matter. Our UIA support should ignore the option in the 
case of Word 2013, as I understand things (which is why I asked in my earlier 
comment).

If we know that UIA in Word 2013 doesn't work remotely well, we should never use 
it. The fact that we were trying to use it in this case is concerning.

**Comment by Adriani90** [comment]
It is not possible to completely exclude a setting from the gui for a certain old version of software. It is only possible to adjust how the default behavior is enforced. And currently you assume the initial author had the default setting value which is just your assumption and we don‘t know if this was the case or not. So let‘s wait feedback from the impacted user as requested above and then see. Von meinem iPhone gesendetAm 24.02.2024 um 02:22 schrieb Luke Davis ***@***.***>:﻿
@Adriani90 it shouldn't matter. Our UIA support should ignore the option in the 
case of Word 2013, as I understand things (which is why I asked in my earlier 
comment).

If we know that UIA in Word 2013 doesn't work remotely well, we should never use 
it. The fact that we were trying to use it in this case is concerning.

**Comment by XLTechie** [comment]
You misunderstand me. I'm not saying we should disappear the setting for such 
users. I'm saying that internally we should ignore the value of the setting, and 
always operate as if it is in the most disabled state it can be. We are most 
certainly capable of doing that.

**Comment by Adriani90** [comment]
@XLTechie I would let NV Access to decide if they want to invest time in investigating and reopening this, I am not sure we should do it for such old software. But maybe @seanbudd or @gerald-hartig can say something about it.
If the default value is not enforced correctly, then I agree we should reopen this. But this is not clear as of now.

**Comment by gerald-hartig** [comment]
@DanielVisionIreland While we prioritise compatibility with current software versions, we'd still like to clarify and potentially improve NVDA's behavior in these situations.

As other people have already suggested on this issue, could you please confirm whether the issue occurs with NVDA's default settings, or if the UIA setting was previously changed? Please set the logging level to "debug" in NVDA's general settings, reproduce the problem, and attach the NVDA log file to this issue.

This will help us determine if there's a bug in how NVDA chooses the appropriate accessibility method for older Office versions, and prioritise the investigation and fix. We think it is worth investigating this issue further if there is evidence that the logic for UIA engagement in Office applications is not functioning as intended. Even if a comprehensive fix for Word 2013 isn't feasible, we may be able to improve NVDA's behaviour for other users who require older versions of Word.

**Comment by DanielVisionIreland** [comment]
> @lukaszgo1 support for Office 2013 ended completely in April 2023: https://support.microsoft.com/en-gb/office/end-of-support-for-office-2013-90e4b0d1-098f-4656-b6e7-8b13b67ed62f
> 
> In Office 2016 I cannot reproduce this problem. Do you still think it is worthwhile investigating? @DanielVisionIreland what is the reason you still stay with Office 2013 instead of updating to 2016 at least?

Hi, it's not my personal decision to go up to a newer version of Office, I would if I could but it's out of my hands unfortunately.

**Comment by Adriani90** [comment]
@DanielVisionIreland if you change now the UIA setting value back to the default value (default, where suitable), are you still seeing this issue?
I am just trying to find out if this is a one time thing or maybe also related to #13704. In Office 2016 it seems to occur only when restarting NVDA while focusing a document by using ctrl+alt+n or nvda+q and restart. But in Office 2013 might be worse indeed.

**Comment by gerald-hartig** [comment]
@DanielVisionIreland thanks for that. Log file is always helpful.

**Comment by DanielVisionIreland** [comment]
> @DanielVisionIreland if you change now the UIA setting value back to the default value (default, where suitable), are you still seeing this issue? I am just trying to find out if this is a one time thing or maybe also related to #13704. In Office 2016 it seems to occur only when restarting NVDA while focusing a document by using ctrl+alt+n or nvda+q and restart. But in Office 2013 might be worse indeed.

----------------
Hi, when I switch the UIA back to default, the problem does not return!
I attach two log files as requested...
[nvda log with UIA default.txt](https://github.com/nvaccess/nvda/files/14447684/nvda.log.with.UIA.default.txt)
[nvda log with UIA only when Neccessary.txt](https://github.com/nvaccess/nvda/files/14447687/nvda.log.with.UIA.only.when.Neccessary.txt)

**Comment by Adriani90** [comment]
Ok it seems it was a one time issue and it disappears when changing the UIA setting back to default. The issue in #13704 is still valid, so I am closing in favor of that one which is clearly reproducible.
